defaults/main.yml

```diff
@@ -35,6 +35,15 @@ __network_packages_default_wireless: ["{%
       if __network_wireless_connections_defined
       %}NetworkManager-wifi{% endif %}"]
 
+# check if any team connections are defined
+__network_team_connections_defined: "{{
+      ['team'] in network_connections|json_query('[*][type]') }}"
```

Comment:
is the value you are looking for in `network_connections` the string value `"team"` or an array/list with a single element with the string value `"team"`?

Reply:
I think it is the string value `team`, the following code extracted from `./library/network_connections.py`
```
        elif connection["type"] == "team":
            ifcfg["DEVICETYPE"] = "Team"
        elif connection["type"] == "vlan":
            ifcfg["VLAN"] = "yes"
            ifcfg["TYPE"] = "Vlan"
```

Reply:
`network_connections` is a list of dictionaries and the dictionaries might contain the `type` key. If there is at least one dictionary with the value `team` for the type, then the variable should be true.

Reply:
The way the code is currently written is looking for an array with a single value of the string `"team"` in the `type` field.  That is, something like this in python:
```python
        elif connection["type"] == ["team"]:
```
Unless I misunderstand what `['team'] in network_connections|json_query('[*][type]')` is supposed to do - that is, did you mean
`'team' in network_connections|json_query('[*][type]')`
?

Reply:
Looks to me that you are right. @ffmancera proposed to create a test to check this, seems like the best approach here.

Reply:
I've tried this out and @liangwen12year too. It seems it is working. But it seems in this context, this `["wireless"]` is useless as `"wireless"` has the same effect and it is clearer. :-)

Reply:
For qe test with `tests_wireless_nm.yml`, we need to use CentOS-7. Please check, https://github.com/linux-system-roles/network/blob/main/tests/tests_wireless_nm.yml#L24.  Hi @ffmancera , I tested with `wireless` (bracket removed), I found the qe test failed, and `NetworkManager-wifi` is not installed in the virtual machine. Hence, we do need to code it as `[wireless]`(bracket included). And the errors attached below: 
```
TASK [linux-system-roles.network : Configure networking connection profiles] *****************************************************************************************************************************************************************
[WARNING]: [003] <info>  #0, state:up persistent_state:present, 'wlan0': add connection wlan0, 47fa7561-1a21-4541-8243-d0e6bd5f33d2
[WARNING]: [004] <info>  #0, state:up persistent_state:present, 'wlan0': up connection wlan0, 47fa7561-1a21-4541-8243-d0e6bd5f33d2 (not-active)
[WARNING]: [005] <error> #0, state:up persistent_state:present, 'wlan0': up connection failed: failure to activate connection: nm-manager-error-quark: No suitable device found for this connection (device lo not available because device
is strictly unmanaged). (3)
fatal: [./CentOS-7-x86_64-GenericCloud-2003.qcow2]: FAILED! => {"_invocation": {"module_args": {"__debug_flags": "", "connections": [{"ip": {"address": ["203.0.113.2/24"], "auto6": "no", "dhcp4": "no"}, "name": "wlan0", "state": "up", "type": "wireless", "wireless": {"key_mgmt": "wpa-psk", "password": "p@55w0rD", "ssid": "mock_wifi"}}], "force_state_change": false, "ignore_errors": false, "provider": "nm"}}, "changed": true, "msg": "error: up connection failed: failure to activate connection: nm-manager-error-quark: No suitable device found for this connection (device lo not available because device is strictly unmanaged). (3)", "stderr": "[003] <info>  #0, state:up persistent_state:present, 'wlan0': add connection wlan0, 47fa7561-1a21-4541-8243-d0e6bd5f33d2\n[004] <info>  #0, state:up persistent_state:present, 'wlan0': up connection wlan0, 47fa7561-1a21-4541-8243-d0e6bd5f33d2 (not-active)\n[005] <error> #0, state:up persistent_state:present, 'wlan0': up connection failed: failure to activate connection: nm-manager-error-quark: No suitable device found for this connection (device lo not available because device is strictly unmanaged). (3)\n", "stderr_lines": ["[003] <info>  #0, state:up persistent_state:present, 'wlan0': add connection wlan0, 47fa7561-1a21-4541-8243-d0e6bd5f33d2", "[004] <info>  #0, state:up persistent_state:present, 'wlan0': up connection wlan0, 47fa7561-1a21-4541-8243-d0e6bd5f33d2 (not-active)", "[005] <error> #0, state:up persistent_state:present, 'wlan0': up connection failed: failure to activate connection: nm-manager-error-quark: No suitable device found for this connection (device lo not available because device is strictly unmanaged). (3)"]}
```

Reply:
Nice, thank you! @liangwen12year